### PR TITLE
fix: support schema different from the default one

### PR DESCRIPTION
**Pull request by ghusse**
Linked to CU-860rr05qh

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-sql/src/introspection/helpers/sql-type-converter.ts [hunk 1]
```diff
@@ -79,10 +82,18 @@ export default class SqlTypeConverter {
...
         'TABLE' = e.object_type AND
+        (:schema IS NULL OR c.table_schema = :schema) AND
         c.dtd_identifier = e.collection_type_identifier
       )
       WHERE table_name = :tableName AND c.column_name = :columnName;`.replace(/\s+/g, ' '),
-      { replacements: { tableName, columnName }, type: QueryTypes.SELECT },
+      {
+        replacements: {
+          tableName: tableIdentifier.tableName,
+          schema: tableIdentifier.schema || null,
```
Comment (Thenkei) [inline packages/datasource-sql/src/introspection/helpers/sql-type-converter.ts:92]: 🤾

File: packages/datasource-sql/src/introspection/introspector.ts [hunk 2]
```diff
@@ -18,45 +23,51 @@ export default class Introspector {
...
     const names: ({ tableName: string } | string)[] = await sequelize
       .getQueryInterface()
       .showAllTables();
 
-    // Fixes Sequelize behavior incorrectly implemented.
-    // Types indicate that showAllTables() should return a list of string, but it
-    // returns a list of object for both mariadb & mssql
     // @see https://github.com/sequelize/sequelize/blob/main/src/dialects/mariadb/query.js#L295
-    return names.map(name => (typeof name === 'string' ? name : name?.tableName));
+    return names.map(tableIdentifier =>
```
Comment (Thenkei) [inline packages/datasource-sql/src/introspection/introspector.ts:32]:
```suggestion
    return tableIdentifiers.map(tableIdentifier =>
```
It's weird in the current state. 👀 
Could we add a comment in there?

File: packages/datasource-sql/src/introspection/introspector.ts [hunk 3]
```diff
@@ -18,45 +23,51 @@ export default class Introspector {
   }
 
   /** Get names of all tables in the public schema of the db */
-  private static async getTableNames(sequelize: Sequelize): Promise<string[]> {
+  private static async getTableNames(sequelize: Sequelize): Promise<SequelizeTableIdentifier[]> {
     const names: ({ tableName: string } | string)[] = await sequelize
```
Comment (Thenkei) [inline packages/datasource-sql/src/introspection/introspector.ts:27]: This type is wrong ?

File: packages/datasource-sql/test/datasource-factory.integration.test.ts [hunk 4]
```diff
@@ -25,43 +27,69 @@ describe('SqlDataSourceFactory > Integration', () => {
...
+    CONNECTION_DETAILS.flatMap(connectionDetails => [
+      [
+        connectionDetails.dialect,
+        connectionDetails.username,
+        connectionDetails.password,
+        connectionDetails.host,
+        connectionDetails.port,
+        undefined,
+      ],
+      ...(connectionDetails.supports.schemas
```
Comment (Thenkei) [inline packages/datasource-sql/test/datasource-factory.integration.test.ts:40]: 🤝

File: packages/datasource-sql/test/datasource-factory.integration.test.ts [hunk 5]
```diff
@@ -75,11 +103,14 @@ describe('SqlDataSourceFactory > Integration', () => {
...
         const logger = jest.fn();
 
-        const setupSequelize = await setupDatabaseWithTypes(baseUri, dialect, databaseName);
+        const setupSequelize = await setupDatabaseWithTypes(baseUri, dialect, databaseName, schema);
         const setupModels = setupSequelize.models;
         const attributesMapping = getAttributeMapping(dialect as Dialect);
 
-        const sequelize = await buildSequelizeInstance(`${baseUri}/${databaseName}`, logger);
+        const sequelize = await buildSequelizeInstance(
+          `${baseUri}/${databaseName}?${queryString}`,
```
Comment (Thenkei) [inline packages/datasource-sql/test/datasource-factory.integration.test.ts:111]: 👀 Why you are not using the same pattern everywhere ?
`${queryString && `?${queryString}`}`

File: packages/datasource-sql/test/introspection/helpers/sql-type-converter.integration.test.ts [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+        });
+
+        afterEach(async () => {
+          sequelize?.close();
+        });
+
+        describe.each(
+          SCALAR_TYPES.filter(
+            ([, , dialects]) => !dialects || dialects.includes(connectionDetails.dialect),
+          ),
```
Comment (Thenkei) [inline packages/datasource-sql/test/introspection/helpers/sql-type-converter.integration.test.ts:69]: 👌

File: packages/datasource-sql/test/datasource-factory.integration.test.ts (hunk 5 above)
Comment (ghusse) [reply packages/datasource-sql/test/datasource-factory.integration.test.ts:111]: I think I wrote `queryString && ...` at first, and then realized it was not useful here and I could simply always add the question mark.

File: packages/datasource-sql/src/introspection/introspector.ts (hunk 2 above)
Comment (ghusse) [reply packages/datasource-sql/src/introspection/introspector.ts:32]: I'll rename the variable and add a comment

**Review by Thenkei — APPROVED** [review]

**Pull request merged**